src/systems/filecoin_blockchain/struct/block/block.id

```diff
@@ -40,11 +39,15 @@ type BlockHeader struct {
     //	ComputeUnsignedFingerprint() []
 }
 
+type TxMeta struct {
+    BLSMessages [msg.UnsignedMessage]
+    SECPMessages [msg.SignedMessage]
```

Comment:
These are both AMTs, the values of which are the CIDs of the messages. I'm not sure how best to represent that here, maybe `&[&msg.UnsignedMessage] // array-mapped trie`

Reply:
`&[&msg.UnsignedMessage]` seems right to me, but as @anorth points out above, this may be a rabbit hole we don't want to deal with yet -- there are many places we currently use type T where we really mean &T, and it might be easier to address all of those in a single subsequent PR.

Reply:
This is a place where the distinction between an embedded and referenced object makes a big difference. Implementations disagree about whether some of these things are CIDs or embedded - the spec needs to be clear.

I used `&` as a shorthand for CID that I'd seen elsewhere, but just as happy with `[cid.CID] // AMT<&UnsignedMessage]> array-mapped trie`

Reply:
That is the larger question. If & is a standin for CID that should be explicitly stated in the spec. I have added accordingly.